### PR TITLE
Add ECS application docs

### PR DESCRIPTION
**Pull request by khanhtc1202**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
Documentation for ECS deployment
```

### CODE REVIEW

File: docs/content/en/docs/operator-manual/piped/adding-a-cloud-provider.md [hunk 1]
```diff
@@ -103,3 +103,30 @@ It attempts to retrieve credentials in the following order:
...
+  cloudProviders:
+    - name: ecs-dev
+      type: ECS
+      config:
+        region: {ECS_CLUSTER_REGION}
+        profile: default
+        credentialsFile: {PATH_TO_THE_CREDENTIAL_FILE}
+```
+
+Just same as Lambda applications' credentials methods, there are several ways to authorize Piped agent to enable it performs deployment jobs.
```
Comment (nghialv) [inline docs/content/en/docs/operator-manual/piped/adding-a-cloud-provider.md:125]: nit:  How about this
`Just same as Lambda cloud provider, `

File: docs/content/en/docs/user-guide/configuring-deployment/ecs.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
+---
+title: "ECS"
+linkTitle: "ECS"
+weight: 5
+description: >
+  Specific guide for configuring Amazon ECS deployment.
+---
+
+Deploying a Amazon ECS application requires `taskdef.yaml` and `servicedef.yaml` file placing inside the application directory. Those files contain all configuration for [ECS TaskDefinition](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definitions.html) object and [ECS Service](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs_services.html) object, and will be used by Piped agent while deploy your application/service to ECS cluster.
```
Comment (nghialv) [inline docs/content/en/docs/user-guide/configuring-deployment/ecs.md:9]: nit: "an Amazon ECS"

File: docs/content/en/docs/user-guide/configuring-deployment/ecs.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+If you're not familiar with ECS, you can get examples for those files from [here](/docs/examples/#ecs-applications).
+
+## Quick sync
+
+By default, when the [pipeline](/docs/user-guide/configuration-reference/#ecs-application) was not specified, PipeCD triggers a quick sync deployment for the merged pull request.
+Quick sync for a ECS deployment will roll out the new version and switch all traffic to it.
```
Comment (nghialv) [inline docs/content/en/docs/user-guide/configuring-deployment/ecs.md:16]: nit: "an ECS"

Comment (nghialv) [reply docs/content/en/docs/user-guide/configuring-deployment/ecs.md:16]: nit: "switch all traffic to it immediately."

File: docs/content/en/docs/user-guide/configuring-deployment/ecs.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+kind: ECSApp
+spec:
+  pipeline:
+    stages:
+      # Deploy the workloads of CANARY variant, the number of workload
+      # for CANARY variant is equal to 10% of PRIMARY's workload.
+      # But this is still receiving no traffic.
+      - name: ECS_CANARY_ROLLOUT
+        with:
+          scale: 10
```
Comment (nghialv) [inline docs/content/en/docs/user-guide/configuring-deployment/ecs.md:52]: nit: As commented before, this can be changed to 30.

Comment (khanhtc1202) [reply docs/content/en/docs/user-guide/configuring-deployment/ecs.md]: Addressed on https://github.com/pipe-cd/pipe/pull/2080/commits/b5f3e68d674fc950bd733abcf071688f01034a07 🙏

File: docs/content/en/docs/user-guide/configuring-deployment/ecs.md (hunk 3 above)
Comment (khanhtc1202) [reply docs/content/en/docs/user-guide/configuring-deployment/ecs.md]: Addressed on https://github.com/pipe-cd/pipe/pull/2080/commits/b5f3e68d674fc950bd733abcf071688f01034a07 🙏

File: docs/content/en/docs/operator-manual/piped/adding-a-cloud-provider.md (hunk 1 above)
Comment (khanhtc1202) [reply docs/content/en/docs/operator-manual/piped/adding-a-cloud-provider.md]: Addressed on https://github.com/pipe-cd/pipe/pull/2080/commits/b5f3e68d674fc950bd733abcf071688f01034a07 🙏

File: docs/content/en/docs/user-guide/configuring-deployment/ecs.md (hunk 2 above)
Comment (khanhtc1202) [reply docs/content/en/docs/user-guide/configuring-deployment/ecs.md]: Addressed on https://github.com/pipe-cd/pipe/pull/2080/commits/b5f3e68d674fc950bd733abcf071688f01034a07 🙏

**Pull request merged**